app/src/ui/lib/update-store.ts

```diff
@@ -55,25 +62,20 @@ class UpdateStore {
       this.lastSuccessfulCheck = new Date(lastSuccessfulCheckTime)
     }
 
-    autoUpdater.on('error', this.onAutoUpdaterError)
-    autoUpdater.on('checking-for-update', this.onCheckingForUpdate)
-    autoUpdater.on('update-available', this.onUpdateAvailable)
-    autoUpdater.on('update-not-available', this.onUpdateNotAvailable)
-    autoUpdater.on('update-downloaded', this.onUpdateDownloaded)
-
-    window.addEventListener('beforeunload', () => {
-      autoUpdater.removeListener('error', this.onAutoUpdaterError)
-      autoUpdater.removeListener(
-        'checking-for-update',
-        this.onCheckingForUpdate
-      )
-      autoUpdater.removeListener('update-available', this.onUpdateAvailable)
-      autoUpdater.removeListener(
-        'update-not-available',
-        this.onUpdateNotAvailable
-      )
-      autoUpdater.removeListener('update-downloaded', this.onUpdateDownloaded)
-    })
+    try {
+      onAutoUpdaterError(this.onAutoUpdaterError)
+      onAutoUpdaterCheckingForUpdate(this.onCheckingForUpdate)
+      onAutoUpdaterUpdateAvailable(this.onUpdateAvailable)
+      onAutoUpdaterUpdateNotAvailable(this.onUpdateNotAvailable)
+      onAutoUpdaterUpdateDownloaded(this.onUpdateDownloaded)
+      setupAutoUpdater()
+
+      window.addEventListener('beforeunload', () => {
+        disposeAutoUpdater()
+      })
+    } catch (e) {
+      this.emitError(e)
+    }
```

Comment:
I don't know - That was copied from original renderer code. I didn't dig into why it was. Happy to remove it - especially since the auto updates appears to capture it errors and send them as event.